tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1360,4 +1360,39 @@ public function commentsProvider() : array
             'valid with extra opening brackets' => ['(DC2Type:should((.stop)).before)', 'should((.stop', $currentType],
         ];
     }
+
+    /**
+     * Returns literals that are platform specific escaped.
+     *
+     * @return array
+     */
+    protected function getEscapedLiterals() : array
+    {
+        return [
+            "\\'", // A single quote
+            '\\"', // A double quote
+            '\\n', // A new-line character
+            '\\r', // A carriage return character
+        ];
+    }
+
+    public function testEscapedDefaultValueMustBePreserved() : void
+    {
+        $value = implode('+', $this->getEscapedLiterals());
+
+        $table = new Table('string_escaped_default_value');
+        $table->addColumn('def_string', 'string', array('default' => $value));
+        $table->addColumn('def_foo', 'string');
+        $this->_sm->dropAndCreateTable($table);
+
+        $onlineTable = $this->_sm->listTableDetails('string_escaped_default_value');
+        self::assertSame($value, $onlineTable->getColumn('def_string')->getDefault(), 'should be able introspect the value of default');
+
+        $comparator = new Comparator();
+        self::assertFalse($comparator->diffTable($table, $onlineTable), 'introspected schema should have no differences');
+
+        $this->_conn->insert('string_escaped_default_value', array('def_foo' => 'foo'));
+        $row = $this->_conn->fetchAssoc('SELECT def_string FROM string_escaped_default_value');
+        self::assertSame($value, $row['def_string'], 'inserted default value should be the configured default value');
```

Comment:
I'd move this assertion to a different test case. If the introspection assertion fails, it's still worth checking if the value was properly deployed. These requirements are independent.

Reply:
I feel this would force us to duplicate code for test setup, and double the time needed (table creation is expensive), while those three assert are very closely related together semantically speaking. Are you sure you want to take the performance hit ?

Reply:
It doesn't have to be a performance hit. On setup, you can create one table with 13 fields representing different defaults and create one record with all default values. The first test would go over fields one by one and introspect them. The second test would select each field and see if the value inserted by default is correct. Each failure will clearly indicate the case where something went wrong.

Reply:
Is it correct that this going against [what you said earlier](https://github.com/doctrine/dbal/pull/2850#discussion_r155142197) and `getEscapedLiterals()` should not be a dataProvider anymore ? But instead have a private method like `createTableForDefaultValues()` which is called in two different test cases `testIntrospectDefaultValues()` and `testInsertDefaultValues` ?

Reply:
Fixed too.